org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/util/JRTUtil.java

```diff
@@ -235,16 +229,11 @@ public static void reset() {
 	 * @param visitor an instance of JrtFileVisitor to be notified of the entries in the JRT image.
 	 * @param notify flag indicating the notifications the client is interested in.
 	 */
-	public static void walkModuleImage(File image, final JRTUtil.JrtFileVisitor<Path> visitor, int notify) throws IOException {
-		JrtFileSystem system = getJrtSystem(image, null);
-		if (system == null) {
-			return;
-		}
-		system.walkModuleImage(visitor, notify);
+	public static void walkModuleImage(File image, JRTUtil.JrtFileVisitor<Path> visitor, int notify) throws IOException {
+		walkModuleImage( getJrtSystem(image, null), visitor, notify);
 	}
 
-	public static void walkModuleImage(File image, String release, final JRTUtil.JrtFileVisitor<Path> visitor, int notify) throws IOException {
-		JrtFileSystem system = getJrtSystem(image, release);
+	public static void walkModuleImage(JrtFileSystem system, JRTUtil.JrtFileVisitor<Path> visitor, int notify) throws IOException {
```

Comment:
Is it really necessary to keep these static methods that just call a similarly named method on the first argument here and below?
Some handle the case where the `JRTFileSystem` is null, but I wonder if that would be better handled by the caller?
And some don't even do null-checks and just call the corresponding method on `JRTFileSystem`.

Reply:
I suggest you answer that question yourself by looking into the call hierarchy.

Reply:
If I do that I see that one just has to change the visibility of the called method to public and then the same question arises, plus should that method not be public (together with the corresponding other methods)?

But in general, since I patiently answer your question and address your remarks on my PRs, I was hoping that you would return the favor and answer my questions/remarks on your PRs.

Reply:
There are 4 callers. i don't see a benefit to add the null check to all of them.